reconcile/utils/aws_api_typed/organization.py

```diff
@@ -17,37 +18,62 @@ class AwsOrganizationOU(BaseModel):
     name: str = Field(..., alias="Name")
     children: list["AwsOrganizationOU"] = []
 
-    def find(self, path: str) -> "AwsOrganizationOU":
-        """Return an organizational unit by its path."""
-        name, *rest = path.strip("/").split("/")
-        subs = "/".join(rest)
-        if self.name == name:
-            if not rest:
-                return self
-            for child in self.children:
-                try:
-                    return child.find(subs)
-                except KeyError:
-                    pass
-        raise KeyError(f"OU not found: {path}")
+    def locate(
+        self, path: list[str], ignore_case: bool = True
+    ) -> Optional["AwsOrganizationOU"]:
+        name, *sub = path
+        match = self.name.lower() == name.lower() if ignore_case else self.name == name
+        if not match:
+            return None
+        if not sub:
+            return self
+        return next(
+            (
+                result
+                for child in self.children
+                if (result := child.locate(sub, ignore_case=ignore_case))
+            ),
+            None,
+        )
+
+    def find(self, path: str, ignore_case: bool = True) -> "AwsOrganizationOU":
+        node = self.locate(path.strip("/").split("/"), ignore_case=ignore_case)
+        if not node:
+            raise KeyError(f"OU not found: {path}")
+        return node
+
+    def __hash__(self) -> int:
+        return hash(self.id)
 
 
 class AWSAccountStatus(BaseModel):
     id: str = Field(..., alias="Id")
-    account_name: str = Field(..., alias="AccountName")
-    account_id: str = Field(..., alias="AccountId")
+    name: str = Field(..., alias="AccountName")
+    uid: str | None = Field(alias="AccountId")
     state: str = Field(..., alias="State")
     failure_reason: CreateAccountFailureReasonType | None = Field(alias="FailureReason")
 
 
+class AWSAccount(BaseModel):
+    name: str = Field(..., alias="Name")
+    uid: str = Field(..., alias="Id")
+    email: str = Field(..., alias="Email")
+    state: str = Field(..., alias="Status")
+
+
 class AWSAccountCreationException(Exception):
-    pass
+    """Exception raised when account creation failed."""
+
+
+class AWSAccountNotFoundException(Exception):
+    """Exception raised when the account cannot be found in the specified OU."""
 
 
 class AWSApiOrganizations:
     def __init__(self, client: OrganizationsClient) -> None:
         self.client = client
 
+    @functools.lru_cache(maxsize=None)
```

Comment:
use `lru_cache` on instance method will cause memory leak, as `self` can't be captured, either change it to a static method/function, or use instance level cache, see https://github.com/app-sre/qontract-reconcile/pull/3388

Reply:
Thank you. Changed in [408eb9f](https://github.com/app-sre/qontract-reconcile/pull/4131/commits/408eb9fe0af7e22c9a8864f2251b5232cd505816)

Reply:
https://github.com/app-sre/qontract-reconcile/pull/4131/files#diff-b5cd26f7416a2b43af10ef4414b9a064817773a86eaa6b3dee16665ed832d801R79 need to be removed too.

Reply:
🤦 removed in [1f5a411](https://github.com/app-sre/qontract-reconcile/pull/4131/commits/1f5a411b3ef35c18ed8265417f4799df48a8dcdb)